bofire/strategies/predictives/predictive.py

```diff
@@ -102,17 +104,32 @@ def predict(self, experiments: pd.DataFrame) -> pd.DataFrame:
             experiments=experiments, specs=self.input_preprocessing_specs
         )
         preds, stds = self._predict(transformed)
+        pred_cols, sd_cols = get_column_names(self.domain.outputs)  # type: ignore
         if stds is not None:
             predictions = pd.DataFrame(
-                data=np.hstack((preds, stds)),
-                columns=["%s_pred" % feat.key for feat in self.domain.outputs.get()]
-                + ["%s_sd" % feat.key for feat in self.domain.outputs.get()],
+                data=np.hstack((preds, stds)), columns=pred_cols + sd_cols
             )
         else:
             predictions = pd.DataFrame(
                 data=preds,
-                columns=["%s_pred" % feat.key for feat in self.domain.outputs.get()],
+                columns=pred_cols,
             )
+        for feat in self.domain.outputs.get():
+            if isinstance(feat, CategoricalOutput):
+                predictions.insert(
+                    loc=0,
+                    column=f"{feat.key}_pred",
+                    value=predictions.filter(regex=f"{feat.key}(.*)_prob")
+                    .idxmax(1)
```

Comment:
This is doubled with `surrogte.py`, could we somehow also write a helper function for this and call them in both ocassions, some method called for example `postprocess_categegorical_predictions`? We could place it in `surrogate.py`. What do you think?

Reply:
I will add it to the naming_conventions file. Unless you can tell me how to access the surrogates within the predictives to call a method specified within the surrogate class?